app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala

```diff
@@ -220,97 +216,106 @@ class BusinessEntityController @Inject()(
-            _.reference
-          }
-          journey = b.fold(throw new IllegalStateException("journey should be defined")) {
-            identity
-          }
-          _ <- escService.addMember(
-            UndertakingRef(
-              ref.getOrElse(throw new IllegalStateException("undertakingRef should be defined"))
-            ),
-            BusinessEntity(
-              EORI(journey.eori.value.getOrElse(throw new IllegalStateException("eori should be defined"))),
-              leadEORI = false,
-              Some(
-                ContactDetails(
-                  journey.contact.value.getOrElse(throw new IllegalStateException("contact should be defined")).phone,
-                  journey.contact.value.getOrElse(throw new IllegalStateException("contact should be defined")).mobile
-                )
-              )
-            )
-          )
-        } yield Redirect(routes.BusinessEntityController.getAddBusinessEntity())
-      })
+      )
   }
 
  def editBusinessEntity(eoriEntered: String): Action[AnyContent] = escAuthentication.async { implicit request =>
    implicit val eori111: EORI = request.eoriNumber
 
    for {
-     a <- escService.retrieveUndertaking(eori111)
-     b <- store.put(BusinessEntityJourney.businessEntityJourneyForEori(a, EORI(eoriEntered)))
-   } yield b match {
-     case journey =>
-       Ok(
-         businessEntityCyaPage(
-           eoriEntered,
-           journey.contact.value.getOrElse(throw new IllegalStateException("missing contact details"))
-         )
-       )
+     undertakingOpt <- escService.retrieveUndertaking(eori111)
+     businessEntityJourney <- store.put(BusinessEntityJourney.businessEntityJourneyForEori(undertakingOpt, EORI(eoriEntered)))
+   } yield {
+     val contactDetails = businessEntityJourney.contact.value.getOrElse(handleMissingSessionData("contact details"))
+     Ok(businessEntityCyaPage(eoriEntered, contactDetails))
    }
  }
 
   def getRemoveBusinessEntity(eoriEntered: String): Action[AnyContent] = escAuthentication.async { implicit request =>
     for {
-      a <- escService.retrieveUndertaking(EORI(eoriEntered))
-    } yield a match {
-      case Some(undertaking) => {
-        val bs = undertaking.undertakingBusinessEntity
-          .filter(a => a.businessEntityIdentifier == eoriEntered)
-          .head
-        Ok(removeBusinessPage(removeBusinessForm, bs))
-      }
+      undertakingOpt <- escService.retrieveUndertaking(EORI(eoriEntered))
+    } yield undertakingOpt match {
+      case Some(undertaking) =>
+        val removeBE = undertaking.undertakingBusinessEntity
+          .filter(_.businessEntityIdentifier == eoriEntered)
+          .headOption.getOrElse(handleMissingSessionData("EORI to be removed"))
+        Ok(removeBusinessPage(removeBusinessForm, removeBE))
+
       case _ => handleMissingSessionData("Undertaking journey")
     }
   }
 
+  def getRemoveYourselfBE: Action[AnyContent] = escAuthentication.async { implicit request =>
```

Comment:
This method name is confusing to me. Is it possible to rename this to something a bit more descriptive?

Reply:
yeah just didn't ant to write Business Entity. I can change that too